### PR TITLE
Fixes: P5-98 - Description of OIL files.

### PR DESCRIPTION
**Pull request by Moesgaarda**

### CODE REVIEW

File: report/3.Theory/OIL.tex [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+\subsubsection*{Status}
+Status specifies the level of error checking, which can be set to either \textit{standard} or \textit{extended}, where extended allows for additional return values for tasks, such as \textit{task is invalid} or \textit{task still occupies resources} \cite{irisa223}.\\
+
+\subsubsection*{Hook routines}
+The OS has a series of boolean hook routines, meaning that they can either be enabled or disabled. 
+\textbf{STARTUPHOOKS} is the first routine to be called with the OS starts, where the user can place initialization procedures, during this hook routine all user interrupts are disabled \cite{irisa223}.
+The \textbf{ERRORHOOK} routine is called in case of application errors, with the purpose of treating status information in a centralized manner. 
+The \textbf{SHUTDOWNHOOK} routine is called by the operating system when the OS service \textit{ShutdownOS} has been called, meaning during the shutdown of the system. 
+The two task routines, \textbf{PRETASKHOOK} and \textbf{POSTTASKHOOK} are called on task context switches, meaning that they can be used for debugging or time measurement, including context switch time \cite{irisa223}.
```
Comment (nillas12) [inline report/3.Theory/OIL.tex]: Generally, maybe for the task context switches and other "special" terms, maybe emphasize them a bit, like the other ones. Dunno, they stick out to me a bit. (task context switches and context switch time)

File: report/3.Theory/OIL.tex [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+As implied by the names, the \textit{PostTaskHook} is called each time before the task leaves the \textit{RUNNING} state; likewise the \textit{PreTaskHook} is called directly before a new task enters the \textit{RUNNING} state\cite{irisa223}.
```
Comment (nillas12) [inline report/3.Theory/OIL.tex]: Shouldn't the ";" just be a "."?

File: report/3.Theory/OIL.tex (hunk 1 above)
Comment (Moesgaarda) [reply report/3.Theory/OIL.tex]: I'm thinking that it was covered last semester, so it should be prior knowledge

File: report/3.Theory/OIL.tex (hunk 2 above)
Comment (Moesgaarda) [reply report/3.Theory/OIL.tex]: I'm pretty sure a semicolon can be used since it's two independent clauses

**Pull request merged**